Index: hypre/src/config/configure.in
===================================================================
--- hypre.orig/src/config/configure.in	2021-05-29 18:37:49.147000455 +0200
+++ hypre/src/config/configure.in	2021-05-29 18:37:49.139000192 +0200
@@ -72,7 +72,7 @@
 
 dnl * Change default prefix from /usr/local to ./hypre
 dnl * Note that $ expansion of shell variables doesn't work with this AC macro.
-AC_PREFIX_DEFAULT([M4_HYPRE_SRCDIR/hypre])
+##AC_PREFIX_DEFAULT([M4_HYPRE_SRCDIR/hypre])
 
 dnl *********************************************************************
 dnl * Initialize some variables
@@ -194,19 +194,18 @@
 dnl *********************************************************************
 dnl * Determine BUILD, HOST, and TARGET types
 dnl *********************************************************************
-if test "x$build_alias" = "x"
-then
-   AC_CANONICAL_BUILD
-fi
+AC_CANONICAL_HOST
+AC_CANONICAL_BUILD
+AC_CANONICAL_TARGET
 
 if test "x$host_alias" = "x"
 then
-   AC_CANONICAL_HOST
+   host_alias=$host
 fi
 
 if test "x$target_alias" = "x"
 then
-   target_alias=$host_alias
+   target_alias=$host
 fi
 
 dnl *********************************************************************
@@ -2370,7 +2369,7 @@
 dnl *********************************************************************
 HYPRE_INSTALLDIR="${prefix}"
 HYPRE_LIBINSTALL="${libdir}"
-HYPRE_INCINSTALL="${includedir}"
+HYPRE_INCINSTALL="${includedir}/hypre"
 
 dnl *********************************************************************
 dnl * Define machine architecture
Index: hypre/src/config/update.sh
===================================================================
--- hypre.orig/src/config/update.sh	2021-05-29 12:19:31.936505065 +0200
+++ hypre/src/config/update.sh	2021-05-29 18:38:15.755863705 +0200
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Copyright 1998-2019 Lawrence Livermore National Security, LLC and other
 # HYPRE Project Developers. See the top-level COPYRIGHT file for details.
 #
